src/modules/exchangeProvider.js

```diff
@@ -94,13 +102,14 @@ const onCancelledOrder = (orderId, maker, tokenAmount, weiAmount) => {
 };
 
 // OrderFill(address indexed tokenBuyer, address indexed tokenSeller, uint buyTokenOrderId, uint sellTokenOrderId, uint price, uint weiAmount, uint tokenAmount);
-const onOrderFill = (tokenBuyer, tokenSeller, buyTokenOrderId, sellTokenOrderId, price, weiAmount, tokenAmount) => {
+const onOrderFill = function(tokenBuyer, tokenSeller, buyTokenOrderId, sellTokenOrderId, price, weiAmount, tokenAmount) {
     console.debug("exchangeProvider.onOrderFill: dispatching refreshExchange() and regreshOrders()");
     // FIXME: shouldn't do refresh for each orderFill event becuase multiple orderFills emmitted for one new order
     //          but newOrder is not emmited when a sell fully covered by orders and
     store.dispatch(refreshExchange());
     store.dispatch(refreshOrders());
     const userAccount = store.getState().web3Connect.userAccount;
+    store.dispatch(processNewTrade('OrderFill', userAccount, this));
```

Comment:
same here, only dispatch if userAccount === maker or userAccount === taker
if you dispatch twice when maker === taker, then it will show up twice in trades history (which is expected if the match is with myself)